pkg/codegen/dotnet/gen_program.go

```diff
@@ -307,16 +307,40 @@ func GenerateProject(
 
 	csproj.WriteString("	<ItemGroup>\n")
 
-	// Add the Pulumi package reference
-	if path, has := localDependencies[pulumiPackage]; has {
-		filename := filepath.Base(path)
-		pkg, rest, _ := strings.Cut(filename, ".")
-		version, _ := strings.CutSuffix(rest, ".nupkg")
-
-		csproj.WriteString(fmt.Sprintf(
-			"		<PackageReference Include=\"%s\" Version=\"%s\" />\n",
-			pkg, version))
-	} else {
+	// Add local package references
+	for _, path := range localDependencies {
+		nugetFile := ""
+		if strings.HasSuffix(path, ".nupkg") {
+			nugetFile = path
+		} else {
+			files, err := os.ReadDir(path)
+			if err != nil {
+				return fmt.Errorf("could not read directory: %w", err)
+			}
+
+			for _, file := range files {
+				if strings.HasSuffix(file.Name(), ".nupkg") {
+					nugetFile = filepath.Join(path, file.Name())
+					break
+				}
+			}
+		}
+
+		filename := filepath.Base(nugetFile)
+		parts := strings.Split(filename, ".")
+		if len(parts) >= 5 {
```

Comment:
Can this logic be shared with a helper method?

Reply:
sure thing